addons/utm/models/utm.py

```diff
@@ -42,14 +49,88 @@ def _group_expand_stage_ids(self, stages, domain, order):
         stage_ids = stages._search([], order=order, access_rights_uid=SUPERUSER_ID)
         return stages.browse(stage_ids)
 
+    def _merge_utm_campaigns(self, campaign=False, name=False, user_id=False, stage_id=False, tag_ids=False):
```

Comment:
I would force the "campaign to keep" as first parameter, as you always pass one anyway.
The fact that you use the first one if not is a bit too "magic" and not really useful.

Reply:
Actually, that's not the case. I don't ask the user to choose a campaign to keep, they can choose nothing. So I add this "magic" part. But I can change it to what you like.